File: typescript/test/helpers/fake-logger.ts
// Drop-in replacement for the SDK's Logger: records every entry passed to
// .log() into an in-memory array so tests can assert on the payload. Cast
// the returned object to `Logger` at the call site — the wraps only ever
// invoke `.log(entry)`, so structural compatibility is enough.
import type { SdkLogEntry } from "../../src/logger.js";

export interface FakeLogger {
  entries: SdkLogEntry[];
  log: (entry: SdkLogEntry) => void;
  flush: () => Promise<void>;
}

export function fakeLogger(): FakeLogger {
  const entries: SdkLogEntry[] = [];
  return {
    entries,
    log(entry) {
      entries.push(entry);
    },
    async flush() {
      // No-op; the real Logger awaits inflight POSTs, but the fake is sync.
    },
  };
}

File: python/tests/helpers/fake_logger.py
"""Drop-in Logger replacement: captures every entry in-memory."""

from __future__ import annotations

from typing import Any


class FakeLogger:
    """Structural compatibility with spaturzu._logger.Logger.

    The wraps only ever call ``.log(entry)``; ``flush`` / ``close`` are
    no-ops here. Tests inspect ``.entries`` to assert on the payload.
    """

    def __init__(self) -> None:
        self.entries: list[dict[str, Any]] = []

    def log(self, entry: dict[str, Any]) -> None:
        self.entries.append(entry)

    def flush(self, timeout_s: float | None = None) -> None:  # pragma: no cover
        pass

    def close(self) -> None:  # pragma: no cover
        pass
